builtin/providers/aws/resource_aws_ami.go

```diff
@@ -196,7 +196,9 @@ func resourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("ebs_block_device", ebsBlockDevs)
 	d.Set("ephemeral_block_device", ephemeralBlockDevs)
 
-	d.Set("tags", tagsToMap(image.Tags))
+	if _, ok := d.GetOk("tags"); ok {
+		d.Set("tags", tagsToMap(image.Tags))
```

Comment:
if I'm not mistaken, only setting the tags if they're found in the config will prevent Terraform from detecting drift for any resource, e.g. if a resource has no tags in the config, but tags are added externally, Terraform would not show a plan/diff to remove the tags.